packages/web-app-files/src/helpers/resource/resource.ts

```diff
@@ -16,9 +16,8 @@ export const extractNameWithoutExtension = (resource?: Resource): string => {
   return name.substring(0, extensionIndexInName)
 }
 
-export const extractExtensionFromFile = (resource: Resource): string => {
-  const name = resource.name
-  if (resource.type === 'dir' || resource.type === 'folder' || resource.isFolder) return ''
+export const extractExtensionFromFile = (name: string, type: string, isFolder: boolean): string => {
```

Comment:
Uff, to be honest I liked to only provide an object. What's your concern with it (genuinely interested)? If it's the fact that the `Resource` interface is too broad, you could introduce an interface `ExtractExtensionFromFileOptions` where you only define the attributes needed for the function and use that as type for the input param of the function. The `Resource` interface would then fulfill that options interface as well so that we could still pass in the resource objects.

Reply:
The issue I have is that the given `resource` is not always of the same type. In `buildResource`, the resource object is the one that comes from the SDK (before it's built). In `renameResource` e.g., it's the built resource object, which is completely different.

Now we mainly need the name of the resource object. That would be the `name` property of the built resource object. For the "unbuilt" object, it's something completely different (parsed path or the new DAV property).

We could introduce `ExtractExtensionFromFileOptions`. But it would have a few optional params because again, the resource object completely differs depending on where the method is called from.

Reply:
The nice thing about objects as params is that you can just omit the attributes that are not relevant for your context (as long as they are optional in the interface of course). So I'd still favour an object as param. More specific interface as discussed is definitely a good idea.